src/EfficientDynamoDb/Internal/Operations/ExecuteTransaction/ExecuteTransactionResponseParser.cs

```diff
@@ -0,0 +1,28 @@
+﻿using EfficientDynamoDb.DocumentModel;
+using EfficientDynamoDb.Internal.Operations.Shared;
+using EfficientDynamoDb.Operations.ExecuteTransaction;
+using System;
+using System.Collections.Generic;
+
+namespace EfficientDynamoDb.Internal.Operations.ExecuteTransaction
+{
+    internal static class ExecuteTransactionResponseParser
+    {
+        public static ExecuteTransactionResponse Parse(Document response)
+        {
+            var responsesArray = response.TryGetValue("Responses", out var responsesAttribute)
+                ? (IReadOnlyList<AttributeValue>)responsesAttribute.AsListAttribute().Items
+                : Array.Empty<AttributeValue>();
+
+            var items = new Document[responsesArray.Count];
+            for (var i = 0; i < responsesArray.Count; i++)
+                items[i] = responsesArray[i].AsDocument()["Item"].AsDocument();
```

Comment:
There is the same set of edge cases as in Batch API. So we need to handle it in the same way.

Reply:
Reviewing the code of the DynamoDBv2 SDK, I see that the ExecuteTransactionResponse class has a List<[ItemResponse](https://docs.aws.amazon.com/es_es/amazondynamodb/latest/APIReference/API_ItemResponse.html)>. Therefore, the ExecuteTransactionResponse class in EfficientDynamoDb would look like this:

```csharp
public class ExecuteTransactionResponse
{
    public IReadOnlyList<ItemResponse> Responses { get; set; } = null!;

    public IReadOnlyList<FullConsumedCapacity>? ConsumedCapacity { get; set; }
}

public class ItemResponse
{
    public Document? Item { get; set; }
}
```
What do you think about this? @firenero

Reply:
Right, all statements must succeed for the transaction to succeed so there is no need for separate errors. Good catch, let's continue with this API.